pkg/tenant/db/tenants.go

```diff
@@ -67,8 +67,9 @@ func RetrieveTenant(ctx context.Context, id uuid.UUID) (tenant *Tenant, err erro
 }
 
 func UpdateTenant(ctx context.Context, tenant *Tenant) (err error) {
-	if tenant.ID == uuid.Nil {
-		return ErrMissingID
+	if tenant.ID.Compare(ulid.ULID{}) == 0 {
+		// TODO: use crypto rand and monotonic entropy with ulid.New
+		tenant.ID = ulid.Make()
```

Comment:
I think we still want to return `ErrMissingID` here on update; otherwise we will be creating a new Tenant record. 

```suggestion
		return ErrMissingID
```